hoomd/md/ForceCompositeGPU.cu

```diff
@@ -713,6 +716,45 @@ __global__ void gpu_update_composite_kernel(unsigned int N,
     quat<Scalar> local_orientation(d_body_orientation[body_indexer(body_type, idx_in_body)]);
     quat<Scalar> updated_orientation = orientation * local_orientation;
 
+    // update velocity
+    quat<Scalar> angmom(d_angmom[central_idx]);
+    Scalar4 velocity(d_velocity[central_idx]);
+    vec3<Scalar> vel(velocity);
+    vec3<Scalar> I(d_inertia[central_idx]);
+
+    Scalar4 vel_mass = d_velocity[idx];
+    const Scalar mass = vel_mass.w;
+
```

Comment:
Sorry, there are a couple things here that I missed in the CPU code:

* Use `const` for things that don't change. `const`-correctness helps prevent bugs, and it can allow compiler optimizations.
* It's not as strict as Python, but prefer a more descriptive name than I. How about `inertia`?
* I don't think `velocity` is used until after after the `if-else` section, so move it there.
* The mixed use of `velocity` and `vel_mass` is also confusing. `vel_mass` is only used to get `mass`, and `mass` is not used until the write transaction. I would go ahead and just read the mass as a single quantity like is done for the `typeid` when it's needed.

```suggestion
    const quat<Scalar> angmom(d_angmom[central_idx]);
    const vec3<Scalar> inertia(d_inertia[central_idx]);
```